core/src/main/scala/org/apache/spark/metrics/source/StaticSources.scala

```diff
@@ -97,6 +97,12 @@ object HiveCatalogMetrics extends Source {
     MetricRegistry.name("parallelListingJobCount"))
 
   /**
+   * Tracks the total number of cachedDataSourceTables hits.
+   */
+  val METRIC_DATASOUCE_TABLE_CACHE_HITS = metricRegistry.counter(
+    MetricRegistry.name("dataSourceTableCacheHits"))
```

Comment:
That's kind of odd, I'd expect the duplicate table building to cause more file accesses or at least cache hits since we are scanning the filesystem multiple times. Is that not the case?